packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/DataStoreObserveEventStreamHandler.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_datastore
+
+import android.os.Handler
+import android.os.Looper
+import com.amplifyframework.datastore.appsync.SerializedModel
+import io.flutter.plugin.common.EventChannel
+
+class DataStoreObserveEventStreamHandler : EventChannel.StreamHandler {
+    private var eventSink: EventChannel.EventSink? = null
+    override fun onListen(argunents: Any?, sink: EventChannel.EventSink) {
+        eventSink = sink
+    }
+
+    fun sendEvent(flutterEvent: Map<String, Any>) {
+        Handler(Looper.getMainLooper()).post {
```

Comment:
Can you pull out the `Handler(Looper.getMainLooper())` into a class-level `var`? What if `sendEvent` happens often -- it'd be a lot of setup/teardown, maybe?

Reply:
Good idea. Done!